### DISCUSSION TITLE
AGA and 020+ support

### DISCUSSION THREAD

**Discussion by emoon** [dirkwhoffmann/vAmiga#471] [Ideas]
Hi,

For my current project which is an Amiga debugger that is aimed at demo development I'm currently considering using fs-uae or vAmiga.

vAmiga is really ideal for what I want except the lack of 020+ and AGA support. The vAmiga codebase is way cleaner than UAE and I would prefer that, but I also want to be able to support AGA thus I'm waning my options.

I wonder if AGA is something that is planned to be supported at some point?

**Comment by dirkwhoffmann** [comment]
Hi Daniel,

right now, I have no such plans. ECS Denise might come, but AGA is extremely unlikely.

**Comment by emoon** [comment]
Alright. Would you be willing to accept PRs for AGA support? (I'm considering my options here)

**Comment by dirkwhoffmann** [comment]
Hi Daniel, 

it’ll be great if you evaluated an experimental AGA version of vAmiga. This would be an interesting evolution of the vAmiga project. However, because AGA is tricky, I expect bigger architectural changes to become necessary. Therefore, I think it would be best to fork the project and develop vAmiga and vAmigaAGA side by side for a while. Of course, code changes in vAmiga could be merged into vAmigaAGA any time. If vAmigaAGA turns out to work as expected, I would merge back AGA support into the main version at a later stage.

**Reply by emoon** [reply]
Cool. That sounds good to me :) Just to be clear. I'm not sure I'm going to do it yet.
